Core/include/Acts/EventData/detail/ParameterTraits.hpp

```diff
@@ -0,0 +1,176 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2016-2020 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/ParameterDefinitions.hpp"
+
+#include <algorithm>
+#include <cmath>
+
+namespace Acts {
+namespace detail {
+
+/// Traits class for an unrestricted parameter.
+struct UnrestrictedParameterTraits {
+  /// Values need no adjustment.
+  static constexpr bool may_modify_value = false;
+
+  /// Get the corrected value within the limits. This is a no-op here.
+  template <typename value_t>
+  static constexpr const value_t& getValue(const value_t& value) {
+    return value;
+  }
+  /// Compute the difference between two values.
+  template <typename value_t>
+  static constexpr value_t getDifference(const value_t& lhs,
+                                         const value_t& rhs) {
+    return lhs - rhs;
+  }
+};
+
+/// Traits class for a parameter with a restricted value range.
+///
+/// @tparam limits_t a type with static `lowest()` and `max()` member functions
+///
+/// This parameter type could be useful to describe parameter with physical
+/// meaningful bounds (e.g. radius).
+template <typename limits_t>
+struct RestrictedParameterTraits {
+  /// Parameter values may need adjustment.
+  static constexpr bool may_modify_value = true;
+  /// Lower bound of range.
+  static constexpr double min = limits_t::lowest();
```

Comment:
Why `lowest` and not `min`?

Reply:
To be consistent with [`std::numeric_limits`](https://en.cppreference.com/w/cpp/types/numeric_limits) and (theoretically) allow `limits_t` to be `std::numeric_limits`. `std::numeric_limits::min()` return the smallest positive positive which would not be what we need here.

Reply:
I wasn't aware of that, thanks